test/functional/plugin/pack_spec.lua

```diff
@@ -1203,6 +1204,150 @@ describe('vim.pack', function()
         validate_hover({ 30, 0 }, 'Add version v1.0.0')
         validate_hover({ 31, 0 }, 'Add version v0.4')
         validate_hover({ 32, 0 }, 'Add version 0.3.1')
+
+        -- textDocument/codeAction
+        n.exec_lua(function()
+          -- Mock `vim.ui.select()` which is a default code action selection
+          _G.select_idx = 0
+
+          ---@diagnostic disable-next-line: duplicate-set-field
+          vim.ui.select = function(items, _, on_choice)
+            _G.select_items = items
+            local idx = _G.select_idx
+            if idx > 0 then
+              on_choice(items[idx], idx)
+              -- Minor delay before continue because LSP cmd execution is async
+              vim.wait(10)
+            end
+          end
+        end)
+
+        local ref_lockfile = get_lock_tbl() --- @type vim.pack.Lock
+
+        local function validate_action(pos, action_titles, select_idx)
+          api.nvim_win_set_cursor(0, pos)
+
+          local lines = api.nvim_buf_get_lines(0, 0, -1, false)
+          n.exec_lua(function()
+            _G.select_items = nil
+            _G.select_idx = select_idx
+            vim.lsp.buf.code_action()
+          end)
+          local titles = vim.tbl_map(function(x) --- @param x table
+            return x.action.title
+          end, n.exec_lua('return _G.select_items or {}'))
+          eq(titles, action_titles)
+
+          -- If no action is asked (like via cancel), should not delete lines
+          if select_idx <= 0 then
+            eq(lines, api.nvim_buf_get_lines(0, 0, -1, false))
+          end
+        end
+
+        -- - Should not include "namespace" header as "plugin at cursor"
+        validate_action({ 1, 1 }, {}, 0)
```

Comment:
If these are test assertions let's always use `assert`. We are pretty consistent about that (in some places we use `expect`). And we are consistent about "validate" denoting parameter/input validation (not a test concept).
```suggestion
        assert_action({ 1, 1 }, {}, 0)
```

Reply:
I forget about that, sorry. Will have to do a cleaning pass on the whole `pack_spec.lua`.